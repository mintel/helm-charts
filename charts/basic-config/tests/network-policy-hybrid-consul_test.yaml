suite: Test Hybrid Consul Network Policy
templates:
  - network-policy-hybrid-consul.yaml
release:
  namespace: test-namespace
tests:
  - it: Renders hybrid consul network policy
    set:
      global.name: test-app
      hybridCloud:
        enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app-allow-consul-proxy
      - contains:
          path: spec.ingress
          content:
            from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: hybrid-consul
            ports:
              - port: 20000
                protocol: TCP
