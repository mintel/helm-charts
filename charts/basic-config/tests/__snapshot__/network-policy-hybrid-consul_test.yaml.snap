Renders hybrid consul network policy:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app-allow-consul-proxy
      name: test-app-allow-consul-proxy
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: hybrid-consul
          ports:
            - port: 20000
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
